beacon-chain/rpc/beaconv1/config.go

```diff
@@ -35,3 +49,37 @@ func (bs *Server) GetDepositContract(ctx context.Context, req *ptypes.Empty) (*e
 		},
 	}, nil
 }
+
+func prepareConfigSpec() (map[string]string, error) {
+	data := make(map[string]string)
+	config := *params.BeaconConfig()
+	t := reflect.TypeOf(config)
+	v := reflect.ValueOf(config)
+
+	for i := 0; i < t.NumField(); i++ {
+		tField := t.Field(i)
+		tag, ok := tField.Tag.Lookup("api")
+		if !ok {
+			// Field should not be returned from API.
+			continue
+		}
+
+		vField := v.Field(i)
+		switch vField.Kind() {
+		case reflect.Uint64:
+			data[tag] = strconv.FormatUint(vField.Uint(), 10)
+		case reflect.Slice:
+			data[tag] = hexutil.Encode(vField.Bytes())
+		case reflect.Array:
+			data[tag] = hexutil.Encode(reflect.ValueOf(&config).Elem().Field(i).Slice(0, vField.Len()).Bytes())
+		case reflect.String:
+			data[tag] = vField.String()
+		case reflect.Uint8:
+			data[tag] = hexutil.Encode([]byte{uint8(vField.Uint())})
+		default:
+			return nil, fmt.Errorf("unsupported config field type: %s", vField.Kind().String())
```

Comment:
Cannot be tested as I am not able to change the type of a field during the test.